### ISSUE TITLE
Crash io.grpc.internal.DelayedClientTransport.newStream

### ISSUE THREAD

**Issue by kainjinez** [firebase/firebase-android-sdk#428]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: 3.3.2
  * Firebase Component: firebase-auth:16.1.0, firebase-firestore:18.0.1, firebase-core:16.0.7, crashlytics:2.9.8, firebase-messaging:17.3.4

### [REQUIRED] Step 3: Describe the problem
I received crash below in Crashlytics. It happened with real user, not me so I don't know how do reproduce and idea to fix it. This is error stack traces:
```
Fatal Exception: java.lang.RuntimeException: Internal error in Firestore (0.6.6-dev).
       at com.google.firebase.firestore.util.AsyncQueue.lambda$panic$5(com.google.firebase:firebase-firestore@@18.0.1:379)
       at com.google.firebase.firestore.util.AsyncQueue$$Lambda$5.run(Unknown Source:2)
       at android.os.Handler.handleCallback(Handler.java:789)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6938)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```
```
Caused by java.lang.RuntimeException: java.lang.NullPointerException: Attempt to invoke virtual method 'io.grpc.LoadBalancer$PickResult io.grpc.LoadBalancer$SubchannelPicker.pickSubchannel(io.grpc.LoadBalancer$PickSubchannelArgs)' on a null object reference
       at com.google.firebase.firestore.util.AsyncQueue.lambda$enqueue$3(com.google.firebase:firebase-firestore@@18.0.1:290)
       at com.google.firebase.firestore.util.AsyncQueue$$Lambda$3.run(Unknown Source:4)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedStartFactory.run(com.google.firebase:firebase-firestore@@18.0.1:205)
       at java.lang.Thread.run(Thread.java:764)
```

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'io.grpc.LoadBalancer$PickResult io.grpc.LoadBalancer$SubchannelPicker.pickSubchannel(io.grpc.LoadBalancer$PickSubchannelArgs)' on a null object reference
       at io.grpc.internal.DelayedClientTransport.newStream(DelayedClientTransport.java:149)
       at io.grpc.internal.ClientCallImpl.start(ClientCallImpl.java:245)
       at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1.start(CensusTracingModule.java:392)
       at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1.start(CensusStatsModule.java:673)
       at com.google.firebase.firestore.util.FirestoreChannel.runBidiStreamingRpc(com.google.firebase:firebase-firestore@@18.0.1:136)
       at com.google.firebase.firestore.remote.AbstractStream.start(com.google.firebase:firebase-firestore@@18.0.1:245)
       at com.google.firebase.firestore.remote.WatchStream.start(com.google.firebase:firebase-firestore@@18.0.1:39)
       at com.google.firebase.firestore.remote.RemoteStore.startWatchStream(com.google.firebase:firebase-firestore@@18.0.1:409)
       at com.google.firebase.firestore.remote.RemoteStore.enableNetwork(com.google.firebase:firebase-firestore@@18.0.1:234)
       at com.google.firebase.firestore.remote.RemoteStore.restartNetwork(com.google.firebase:firebase-firestore@@18.0.1:280)
       at com.google.firebase.firestore.remote.RemoteStore.lambda$new$0(com.google.firebase:firebase-firestore@@18.0.1:220)
       at com.google.firebase.firestore.remote.RemoteStore$$Lambda$3.run(Unknown Source:2)
       at com.google.firebase.firestore.util.AsyncQueue.lambda$enqueue$4(com.google.firebase:firebase-firestore@@18.0.1:311)
       at com.google.firebase.firestore.util.AsyncQueue$$Lambda$4.call(Unknown Source:2)
       at com.google.firebase.firestore.util.AsyncQueue.lambda$enqueue$3(com.google.firebase:firebase-firestore@@18.0.1:287)
       at com.google.firebase.firestore.util.AsyncQueue$$Lambda$3.run(Unknown Source:4)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:457)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedStartFactory.run(com.google.firebase:firebase-firestore@@18.0.1:205)
       at java.lang.Thread.run(Thread.java:764)
```
#### Steps to reproduce:

Users said that everytime they open app it will crash immediately. Crashed-app users mostly use Galaxy series from Samsung. Including S8, S9+ and some J-series.
I'm also using S8 but did not face it.
If you have any idea about which type of function i'm using may happen this, i'll posh here. Thanks.
#### Relevant Code:
N/A

**Comment by kukuku0517** [comment]
same issue here

`Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'io.grpc.LoadBalancer$PickResult io.grpc.LoadBalancer$SubchannelPicker.pickSubchannel(io.grpc.LoadBalancer$PickSubchannelArgs)' on a null object reference
       at io.grpc.internal.DelayedClientTransport.newStream(DelayedClientTransport.java:149)
       at io.grpc.internal.ClientCallImpl.start(ClientCallImpl.java:245)
       at io.grpc.internal.CensusTracingModule$TracingClientInterceptor$1.start(CensusTracingModule.java:392)
       at io.grpc.internal.CensusStatsModule$StatsClientInterceptor$1.start(CensusStatsModule.java:673)
       at com.google.firebase.firestore.util.FirestoreChannel.runBidiStreamingRpc(com.google.firebase:firebase-firestore@@18.1.0:136)
       at com.google.firebase.firestore.remote.AbstractStream.start(com.google.firebase:firebase-firestore@@18.1.0:245)
       at com.google.firebase.firestore.remote.WatchStream.start(com.google.firebase:firebase-firestore@@18.1.0:39)
       at com.google.firebase.firestore.remote.AbstractStream.lambda$performBackoff$2(com.google.firebase:firebase-firestore@@18.1.0:402)
       at com.google.firebase.firestore.remote.AbstractStream$$Lambda$2.run(Unknown Source:2)
       at com.google.firebase.firestore.util.ExponentialBackoff.lambda$backoffAndRun$0(com.google.firebase:firebase-firestore@@18.1.0:123)
       at com.google.firebase.firestore.util.ExponentialBackoff$$Lambda$1.run(Unknown Source:4)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedTask.handleDelayElapsed(com.google.firebase:firebase-firestore@@18.1.0:129)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedTask.access$lambda$0(Unknown Source)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedTask$$Lambda$1.run(Unknown Source:2)
       at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:459)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at com.google.firebase.firestore.util.AsyncQueue$DelayedStartFactory.run(com.google.firebase:firebase-firestore@@18.1.0:205)
       at java.lang.Thread.run(Thread.java:764)`

**Comment by kainjinez** [comment]
@rsgowman i see referenced bug was closed. When will Firestore update it?

**Comment by mikelehen** [comment]
@kainjinez We'll need to wait for a new release (https://github.com/grpc/grpc-java/releases) that includes the fix unfortunately...

**Comment by kainjinez** [comment]
@mikelehen Hi Mike. I saw that they committed this fix to newest version 1.21. When will i see it in firebase?

**Comment by rsgowman** [comment]
The update went out in version 19.0.2 (https://firebase.google.com/support/release-notes/android#version_1902) so you should be able to try it out now. Let us know if this is still a problem for you.
